engine/classes/Elgg/DataStorageService.php

```diff
@@ -0,0 +1,40 @@
+<?php
+namespace Elgg;
+use Elgg\Filesystem\Adapter\Adapter;
+
+class DataStorageService {
+	private $logger;
+	private $adapter;
+	
+	// @todo file/dir object cache.
+	
+	
+	public function __construct(Logger $logger) {
+		$this->logger = $logger;
+	}
+	
+	public function setAdapter(Adapter $adapter) {
```

Comment:
I'd recommend just using constructor params instead of setters. We probably don't want the adapter changing mid-request anyways, and this class has no reasonable behavior without the adapter configured.

Actually, I'm not sure I understand what this class is for. The API is no different than the adapters, really.